src/app/rpmostree-override-builtins.cxx

```diff
@@ -86,10 +88,29 @@ handle_override (RPMOSTreeSysroot  *sysroot_proxy,
   g_variant_dict_insert (&dict, "no-overrides", "b", opt_reset_all);
   g_variant_dict_insert (&dict, "initiating-command-line", "s", invocation->command_line);
   g_variant_dict_insert (&dict, "lock-finalization", "b", opt_lock_finalization);
+  g_variant_dict_insert (&dict, "ex-pin-from-repos", "b", opt_ex_pin_from_repos);
   g_autoptr(GVariant) options = g_variant_ref_sink (g_variant_dict_end (&dict));
 
   g_autoptr(GVariant) previous_deployment = rpmostree_os_dup_default_deployment (os_proxy);
 
+  if (override_replace && opt_ex_pin_from_repos)
```

Comment:
I think handling this here makes a lot of sense, though calling `sort_pkgs_strv` here is confusing because e.g. it'll yield fds we are not yet ready to pass on to the daemon.

Maybe instead we can do something like this (semi-pseudo-code):

```
g_autoptr(GPtrArray) override_replace_final = NULL;
if (override_replace && opt_ex_pin_from_repos)
  {
    initialize override_replace_final
    for each arg in override_replace:
        if arg starts with "http://" or "https://" or ends with ".rpm":
          append arg to override_replace_final
          continue # handled later
        
        # it's a repo package; resolve to a URL using Search() D-Bus API
        # and then append URL to override_replace_final

    add a final NULL to override_replace_final
    override_replace = override_replace_final
  }
```